code/api/views.py

```diff
@@ -448,30 +449,55 @@ def post(self, request, author_id, post_id):
             return HttpResponseForbidden()
 
         comment = request.POST.get('comment')
+        post_type = request.POST.get('post_type')
 
-        # check if empty
+        # check if empty comment
         if not len(comment):
             return HttpResponseBadRequest("Comment cannot be empty.")
 
         pub_date = datetime.now(timezone.utc)
 
+        author = get_object_or_404(LocalAuthor, pk=author_id)
+
         try:
-            author = get_object_or_404(LocalAuthor, pk=author_id)
-            post = get_object_or_404(LocalPost, id=post_id)
+            if post_type == "local":
+                post = get_object_or_404(LocalPost, id=post_id)
+
+                # create local comment
+                Comment.objects.create(
+                    author=author,
+                    post=post,
+                    comment=comment,
+                    content_type='PL',  # TODO: add content type
+                    pub_date=pub_date,
+                )
+            elif post_type == "inbox":
+                post = get_object_or_404(InboxPost, id=post_id)
 
-            comment = Comment.objects.create(
-                author=author,
-                post=post,
-                comment=comment,
-                content_type='PL',  # TODO: add content type
-                pub_date=pub_date,
-            )
+                # create post reqeust data
+                data = {
+                    "@context": "https://www.w3.org/ns/activitystreams",
+                    "summary":f"{author.username} Commented on your post",
+                    "type": "comment",
+                    "author": author.as_json(),
+                    "comment": comment,
+                    "contentType": "text/plain",
+                    "object": post.public_id
+                }
+
+                request_url = f'{post.author.strip("/")}/inbox'
+
+                # send comment to remote inbox
+                api_requests.post(url=request_url, data=data, send_basic_auth_header=True)
+
+            else:
+                HttpResponseNotFound()
 
         except Exception as e:
             logger.error(e, exc_info=True)
             return HttpResponse('Internal Server Error')
 
-        return redirect('socialDistribution:single-post', post_type="local", id=post_id)
+        return redirect('socialDistribution:single-post', post_type=post_type, id=post_id)
```

Comment:
Looks good, but can you please move all the functionality in this function to a function in `socialDistribution/views.py`? This doesn't really make sense here since it redirects to a client view (instead of a typical API JSON response for example).

Reply:
Yes, thanks!